modules/ravestate/context.py

```diff
@@ -168,6 +168,22 @@ def add_prop(self, *, prop: property.PropertyBase) -> None:
             for signalname in self.default_property_signal_names:
                 self.states_per_signal[s(prop.fullname() + signalname)] = set()
 
+    def rm_prop(self, *, prop: property.PropertyBase) -> None:
+        """
+        Remove a property from this context.
+        Generates error message, if the property was not added with add_prop() to the context previously
+        :param prop: The property to remove.object
+        """
+        if prop.fullname() not in self.properties:
+            logger.error(f"Attempt to remove unknown property {prop.fullname()}!")
+            return
+        # remove property from context
+        self.properties.pop(prop.fullname())
+        # remove all of the property's signals
+        with self.states_lock:
```

Comment:
There's a small thing missing here:
We also need to call `rm_state(x)` for all states x that depend on read- or write-access to the property that is about to be deleted.